Source/WarpX.cpp

```diff
@@ -646,12 +652,36 @@ WarpX::ReadParameters ()
             moving_window_v *= PhysConst::c;
         }
 
+        pp_warpx.query("B_ext_grid_init_style", WarpX::B_ext_grid_s);
+        pp_warpx.query("E_ext_grid_init_style", WarpX::E_ext_grid_s);
+
+        if (WarpX::B_ext_grid_s == "read_from_file")
+        {
+            add_external_B_field = true;
+        }
+        if (WarpX::E_ext_grid_s == "read_from_file")
+        {
+            add_external_E_field = true;
+        }
+
         electrostatic_solver_id = GetAlgorithmInteger(pp_warpx, "do_electrostatic");
         // if an electrostatic solver is used, set the Maxwell solver to None
         if (electrostatic_solver_id != ElectrostaticSolverAlgo::None) {
             electromagnetic_solver_id = ElectromagneticSolverAlgo::None;
         }
 
+        if (add_external_B_field) {
+            WARPX_ALWAYS_ASSERT_WITH_MESSAGE(electromagnetic_solver_id == ElectromagneticSolverAlgo::None
+                                             && (electrostatic_solver_id == ElectrostaticSolverAlgo::LabFrame
+                                                 || electrostatic_solver_id == ElectrostaticSolverAlgo::None),
+                                             "External B field is mutually exclusive with B field solution");
+        }
+        if (add_external_E_field) {
+            WARPX_ALWAYS_ASSERT_WITH_MESSAGE(electromagnetic_solver_id == ElectromagneticSolverAlgo::None
+                                             && electrostatic_solver_id == ElectrostaticSolverAlgo::None,
+                                             "External E field is mutually exclusive with E field solution");
+        }
+
```

Comment:
Is this limitation necessary for a reason I'm not seeing? The external field would just set the initial values of the E and B fields, right? It should then be fine to let those initial fields evolve consistently with the particle dynamics, right?